### ISSUE TITLE
DoES should own its own "cloud" infrastructure

### ISSUE THREAD

**Issue by amcewen** [DoESLiverpool/somebody-should#18]
Currently doesliverpool.com is part of my account, wiki.doesliverpool.com is hosted on my server, and the doesliverpool.com website is hosted on @OddEssay's hosting.

We should move to a model where DoES has its own hosting, etc. so more people can update things (and in case anyone gets run over by a bus).

AIMES did offer ages ago to host a VPS for us, so if we wanted we could ask if the offer is still there...

**Comment by DefProc** [comment]
I've recently set up my own gitlab install on a Digital Ocean droplet. As part of moving the wiki to github #97 we will lose the organiser only pages. This could be solved by using the groups feature of gitlab (equivalent to the organisation feature in github). 

This would also allow us to use gitlab as the central space for storing projects (instead of using github) as part of moving to our own infrastructure.

**Comment by paulfurley** [comment]
Not that I've got a great track record of contributing code here so far, but my chance of doing so would go to basically zero if we move off GitHub. Even *Google* have moved their Open Source stuff to GitHub!

Maybe it's just me, but if something's open source and NOT on Github these days I basically either ignore it or mirror it (see https://github.com/paulfurley/gnupg-mirror). Specifically, I love that it's so plainless fork & pull request on Github, using their `hub` command to do so. I'm sure Gitlab has similar features... but...

I think getting site hosting off @amcewen and @OddEssay's own personal machines is a great idea, but to me it's quite a different matter than doing a home-grown source control system.

**Comment by DefProc** [comment]
I've been quite impressed with gitlab (as it's largely a self hosted 
clone of github) that I've moved my internal projects into it (although 
I've been slowly moving to storing all client info in git, even meeting 
minutes).

e.g. https://git.defproc.co.uk/DefProc/thesis

If the option for non-public git repos is either paying github or 
self-hosting a github-alike install, which best fits the suggestion 
about DoES owning it's own resources? (especially considering git is 
distributed anyway!)

**Comment by amcewen** [comment]
Adding the "organisers meeting action" label, as this came up in the last organisers meeting.

**Comment by amcewen** [comment]
The wiki has mostly moved to github now (issue #97), so we're on our way to migrating away from things I have sole control over.

I've just raised a support ticket with 34sp to see about spinning the domains over into a new DoES-only 34sp account, so that's not just under my sole control.

**Comment by jamieisboss** [comment]
After experimenting with digitalocean, google, AWS and AWS through bitnami over the past year or so, we've decided that the best offering for us, which also works out cheapest and has the broadest range of free supporting tech is AWS by a long way. 

- S3 and cloudfront are just there, 
- Lambda doesn't have a comparable service anywhere im aware of, 
- the new container registry is now live, 
- IAM does full user account management 
- you can also use the directory service if you want LDAP
- the aurora database service is ridiculously fast, 
- Quicksight, which i imagine will be launched at this summit, looks like it's a magic autoconf for BI that costs less than having a microsoft rep look in your direction. 

Plus Virtual private cloud is not only free and doesn't seem to have a competitive service elsewhere (though i haven't tried Azure), but it would allow for their IOT  and monitoring services to interact with your IOT devices, doorbots, wifi, anything really, via a VPN into an on demand infrastructure that noone online could see or access from outside DoES without a vpn, unless you enabled it.. That's a lot of Buzzwords I know , but i've sent a lot of time comparing platforms this year.

While i was doing that, amazon launched 7 new services, 2 of which were free and one actually means they charge you less over time (as of today, it's a feature, S3 lifecycle services, unused files automatically get moved to slower cheaper services after a policy period you define, eventually hitting glacier which has a 4 hour recovery time but costs $0.007 per GB / month.)

Then there's virtual desktops, app steaming, transactional email, message queueing, i'm not just going to list them all but the fact that i haven't covered half and thy have  something that hits nearly every one of the issues on here in some way would sway me.

It did for solve, we're moving everything there. Plus you get a year free to play with it. Also, they're launching new stuff, again, nowish.

**Comment by paulfurley** [comment]
I'll second the AWS with a few caveats:

- learning the whole VPC (virtual private something?!), Subnet, Internet Gateway, Security Groups jargon and stuff is initially painful - amazing for massive orgs but feels like huge overkill for small ones
- lock-in lock-in lock-in - all those lovely (REALLY lovely) products like Simple Queue Service are fabulous once you've configured them, but very difficult to extract yourself from. Perhaps not a problem?

Things I really like:

- RDS, the database-as-service is just brilliant
- even tiny $5 boxes are treated with respect and don't appear to disappear without warning like some other providers
- security is baked in everywhere

**Comment by amcewen** [comment]
I'm in favour of us not tying our infrastructure tightly to a single supplier (where possible).  I don't mind too much who we use apart from that.

**Comment by jamieisboss** [comment]
Agree with the lock in but don't see a way around it other then not using the features that make it worth it other than bearing it in mind and trying nt to rely on them too heavily. Perhaps also assuming (usually not a good idea i know but they re big companies fighting) that similar services will appear with toher providers in time as they compete.

>  learning the whole VPC (virtual private something?!), Subnet, Internet Gateway, Security Groups jargon and stuff is initially painful - amazing for massive orgs but feels like huge overkill for small ones

It's alright if you're a network dude, but they made a workaround to learning with [cloudformer](http://aws.amazon.com/developertools/6460180344805680) which creats templates for cloudformation, which builds everything for you (supposedly, Admittedly i havent used this but will do and reeport back)

||TODO||

**Comment by amcewen** [comment]
The two domains I've got control over (doesliverpool.com and doesliverpool.co.uk) are now both transferred to the DoES 34sp account.

I think that just leaves the organisers-only remnants of the wiki and the weeknotes script as things still running on servers only I have access to.  (And the weeknotes script is in the DoES github account, so it's only the config file that *really* needs stashing somewhere to provide resilience)

**Comment by DefProc** [comment]
@amcewen would putting the config files in a private bitbucket repo solve the *only has access* even if it doesn't yet solve the *own it's own infastructure* requirement yet

**Comment by amcewen** [comment]
Yeah, did think about that, but hadn't dug out my bitbucket details at that point :-)

I've created a new private bitbucket repo in the DoES organisation with the config file in it.

**Comment by johnmckerrell** [comment]
Notes from meeting 20160614 - Mostly done - makernight.co.uk needs to move, doesliverpool.com has moved (but needs further improving)

 - [ ] Move makernight.co.uk domain to 34sp
 - [ ] Move makernight.co.uk web site
 - [ ] Confirm that status.doesliverpool.com can be replicated as it's useful to be hosted elsewhere
 - [ ] Organisers pages from wiki.doesliverpool.com  (#299)

**Comment by madjake45** [comment]
With the status.doesliverpool.com all you need to do is install nodejs run
npm install and then set the api keys in the config and your are ready to go

On 14 June 2016 at 14:00, John McKerrell <notifications@github.com> wrote:

> Notes from meeting 20160614 - Mostly done - makernight.co.uk needs to
> move, doesliverpool.com has moved (but needs further improving)
>
>    - Move makernight.co.uk domain to 34sp
>    - Move makernight.co.uk web site
>    - Confirm that status.doesliverpool.com can be replicated as it's
>    useful to be hosted elsewhere
>    - Organisers pages from wiki.doesliverpool.com (#299
>    <https://github.com/DoESLiverpool/somebody-should/issues/299>)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DoESLiverpool/somebody-should/issues/18#issuecomment-225873489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGVbX55TQtkvCSd7k_o0rPsNOi_GNwsYks5qLqYAgaJpZM4Chth0>
> .
>



-- 
-------
Jake

**Comment by amcewen** [comment]
Still ongoing.

**Comment by amcewen** [comment]
Ongoing, with some movement (in migrating the wiki pages over)

**Comment by DoESsean** [comment]
Is this issue still ongoing? What is the current state of play?

**Comment by johnmckerrell** [comment]
Stalled on makernight.co.uk I think, I pinged @OddEssay recently but didn't hear back. As it's down at least there won't be people going the wrong place.

**Comment by ajlennon** [comment]
Is there a cloud infrastructure now?

**Comment by johnmckerrell** [comment]
I'm looking at transferring makernight.co.uk again, 34sp is in a weird state because I initiated a transfer in December but didn't complete it. @OddEssay has now made the change he needs to make so hopefully we can get it completed, I've pinged 34sp for help. I'd quite like to get the website live again in some way as it's not good to leave dead links lying around!

**Comment by johnmckerrell** [comment]
We now own makernight.co.uk, I think the only remaining service that we don't host ourselves is the status site. @skos-ninja are you able to give someone(s) access to this? Also @amcewen did you ever put the week notes config `somewhere-safe`?

**Comment by amcewen** [comment]
The weeknotes config file was already `somewhere-really-safe`, but I've updated it to the latest version.

**Comment by DoESsean** [comment]
Where are we at with this?

**Comment by RiverMersey** [comment]
Not sure, but there are similar current problems with DoES owning/not owning /having access to services such as Google maps and the images of DoES that Google maps displays on a search. Namely, Google still refers to DoES as being in the Hanover Street building.

**Comment by johnmckerrell** [comment]
@RiverMersey as far as I can tell Google no longer refers to DoES as being on Hanover Street, if you're still seeing that can you confirm where? We do own the Google Maps listing now but unfortunately have little control over user photos. If we can't get the ones I've flagged removed then we might want to close the existing location and add a new one, which is perhaps what we should have done originally.

**Comment by RiverMersey** [comment]
@johnmckerrell , yes, what I mean is that when visiting does Liverpool on google maps, many of the photos attached bring-up a mini-map in their lower left corner of Hanover Street

**Comment by amcewen** [comment]
Given we've got a Digital Ocean droplet running for the #670, maybe we can set up a docker install for `status.doesliverpool.com` and run it from there?  Then we could close this newest-oldest issue :laughing:

**Comment by ajlennon** [comment]
How much is that? I've had to use them last year and I am unimpressed. Costly and no export facility.

Hetzner server auction that PaulF put me onto has been a fantastic experience. Lovely heavy duty server for < £25 a month, no tie ins, exceptional administrative support. I have proxmox on mine thanks to @MatthewCroughan and MethodDan and I couldn't wish for a better more flexible low cost setup.

**Comment by MatthewCroughan** [comment]
@ajlennon It will cost way less than £25/month since it's a VPS/Droplet and not a beefy dedicated server like what you have. However, Hetzner do offer even lower cost VPS' than Digital Ocean droplet ($5+ per month at the moment) at € 2.99~ a month.

Sources:
https://www.digitalocean.com/pricing/
https://www.hetzner.com/cloud

**Comment by Half-Shot** [comment]
Can confirm that Hetzner are a quality provider. I moved most of my infra from DO to Hetzner because the prices are better, and the featureset is mostly identical.
